### PR TITLE
Fix merging filter chains with matching all conditions

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Fixes: https://github.com/istio/istio/issues/19935

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -2217,7 +2223,7 @@ func mergeTCPFilterChains(incoming []*listener.FilterChain, pluginParams *plugin
 			}
 		}
 
-		if !conflictFound {
+		if !conflictFound && !replacedFallthrough {
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/listener.go]: I am confused by this, why do we not add it if we replaced fallthrough?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: If we have already replaced a wildcard matching fallthrough FC with the incoming one, we will not need to add the incoming one again.

Comment (howardjohn) [reply pilot/pkg/networking/core/v1alpha3/listener.go]: I just read the code wrong, sgtm

**Comment by hzxuzhonghu** [comment]
Will cherry pick to 1.4 manually.

**Pull request merged**